### PR TITLE
1.0.0

### PR DESCRIPTION
**Pull request by f**
## v1.0.0

- A complete rewrite, more extensible.
- Readable and better code.
- Update to Webpack 4
- Remove built-in loaders. Maybe we can create another repository including custom spinners.
- Remove `width` and `height` props.
- Strict props.
- `isLoading` supports `matcher` now.
- Rename `registerComponents` to `registerComponent`
- Added `accessorName` option to change `$vueLoading` key.
- Removed `createActionHelpers`, use `mapLoadingActions` or `wrapLoading` instead.

### CODE REVIEW

**Comment by f** [comment]
@sky-code I'm in rewriting progress. Can you please review?

**Comment by f** [comment]
Diff view is untrackable, please pull and view code on your editor :)

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 
 node_js:
-  - 5
+  - 10
```
Comment (Koc) [inline .travis.yml]: Is it compatible with Node 8?

File: README.md [hunk 2]
```diff
@@ -116,7 +134,15 @@ Returns boolean value if given loader exists in page.
 </template>
 ```
 
-#### `$isLoading(loaders Array<String>)`
+Also you can use RegExp matcher to make it more flexible:
+
+```html
+<template>
+  <progress-bar v-if="$vueLoading.isLoading(/creating.*/)">Creating something...</progress-bar>
```
Comment (Koc) [inline README.md]: Dot requires escaping. WDYT about using wildcards? It simpler and not requires escaping. We can outsource it to library like https://github.com/sindresorhus/matcher

**Comment by Koc** [comment]
Code is much better readable now. Can you fix #28 also before 1.0, please?

File: .travis.yml (hunk 1 above)
Comment (f) [reply .travis.yml]: Possibly yes. Not tested.

**Comment by f** [comment]
I think #28 is fixed with `vuexModuleName` option. Now no named property is hardcoded, everything is configurable.

I'm thinking about renaming the project but `vue-loading` is already registered. Maybe we can find another fancy name.

**Comment by Koc** [comment]
> I think #28 is fixed with `vuexModuleName` option. Now no named property is hardcoded, everything is configurable.

So should I use different `vuexModuleName` for each Vue instance? If yes - it doesn't sove issue, because we need share same module between different instances.

File: src/vue-loading.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      Vue.component(this.options.componentName, vLoading);
+    }
+
+    if (this.options.useVuex) {
+      const { vuexModuleName } = this.options;
+      if (!store) {
+        throw new Error('[vuex-loading] Vuex not initialized.');
+      }
+      this.store = store;
+      store.registerModule(vuexModuleName, vuexStore);
```
Comment (Koc) [inline src/vue-loading.js]: Maybe it is possible to check here - if module with same name already registered and if yes - skip registration? Can you add test case for multiple Vue instances on same page?

Comment (f) [reply src/vue-loading.js]: This makes sense. I'm going to.

**Pull request merged**